ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceWithAssignmentExpressionFix.fs

```diff
@@ -0,0 +1,27 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings.CommonErrors
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+open JetBrains.ReSharper.Psi.Tree
+
+type ReplaceWithAssignmentExpressionFix(warning: UnitTypeExpectedWarning) =
+    inherit FSharpQuickFixBase()
+   
+    let expr = warning.Expr.As<IBinaryAppExpr>()
+    override x.IsAvailable _ =        
+        if not (isValid expr) || expr = null then false else
+        match expr.LeftArgument with
+        | :? IReferenceExpr
+        | :? IIndexerExpr -> isEquals expr.Operator.FirstChild
```

Comment:
Since what you're doing here is a workaround for the grammar rule, could you add a comment explaining the logic?